### DISCUSSION TITLE
lconfig command KeyError

### DISCUSSION THREAD

**Discussion by PaoloLaMuro** [Kidel/Kathara#253] [Q&A] [answered]
Salve, non capisco l'errore in foto. 
Sto tentando di collegare la macchina con wireshark come da guida ma mi restituisce questo errore.
<img width="754" alt="Screenshot 2023-12-05 200355" src="https://github.com/KatharaFramework/Kathara/assets/128410303/70c81699-bf43-4818-ae66-fba41d4de2e7">

**Comment by tcaiazzi** [comment]
Hi @PaoloLaMuro, 

Thanks for opening the discussion.  I need further information to help you:

1. What OS are you using? 
2. Can you share the network scenario and steps to reproduce the error? 

Thanks, 
Tommaso

P.S. Please write in English here so that your error can help other people.

**Comment by PaoloLaMuro** [comment]
thank you for answering.

1.I'm running windows 11 64bit
2. in this scenario there are 2 pc connected with a lan 

this is the lab.conf
LAB_DESCRIPTION="A simple example two computers on a single lan"
LAB_VERSION=1.0
LAB_AUTHOR="L. Ariemma"
LAB_EMAIL=contact@kathara.org
LAB_WEB=http://www.kathara.org/

pc1[0]=A
pc1[image]="kathara/base"
pc1[ipv6]="false"

pc2[0]=A
pc2[image]="kathara/base"
pc2[ipv6]="false"

wireshark[bridged]=true
wireshark[port]="3000:3000/tcp"
wireshark[image]="lscr.io/linuxserver/wireshark"

i'm trying at this step to connect a wireshark device with the comand "kathara lconfig -n wireshark --add A"
 as in the pic.

**Reply by PaoloLaMuro** [reply]
p.s. i've copied it wrong here but i've alredy tried without "\tcp" or with "\tcp".

**Comment by tcaiazzi** [comment]
Hi @PaoloLaMuro, 

I managed to reproduce the error.  Probably you have run the lab, but your containers are stopped (maybe you rebooted your machine?) as in the following image:

![immagine](https://github.com/KatharaFramework/Kathara/assets/26346836/d198b246-d4d4-4008-bb43-a433d24a95bb)


To fix the error, you only need to restart the lab and then connect the `wireshark` device to collision domain `A`: 
```bash 
kathara lrestart 
kathara lconfig -n wireshark --add A
```

Let me know if this helps. 

Cheers, 
Tommaso

**Reply by PaoloLaMuro** [reply]
Thank you Tommaso but still doesen't working. My containers are fully running. I've tried to restart the lab few times, I've tried with different labs, same problem always.
I've just tried your input to be sure and still doesn't work.

**Reply by PaoloLaMuro** [reply]
update, at the end i've solved with a clean re-install of docker. Now it seems to work.

**Reply by tcaiazzi** [reply]
Hi @PaoloLaMuro,

I'm glad that you resolved the problem. 

Do you know what version of Docker are you using before?

**Comment by tcaiazzi** [comment, marked as answer]
Hi @PaoloLaMuro, 

I found the reason for your error.  You can check #256 for more details. 

The steps to reproduce the error are the following:

1. Start a machine called `X` with `vstart`
2. Start a network scenario `Y` containing a machine `X` with `lstart`
3. Now you have two machines called X running, one in the `vlab` and one in `Y`.
4. Run `lconfig` in `Y` attaching another CD to X, Kathará will return you the error:
```bash
CRITICAL - (KeyError) 'kathara_lollo-fx5lx6ymglgwdoljxihqjw_A'
```

We will fix the problem soon.

**Reply by tcaiazzi** [reply]
The problem is fixed!
